variables.tf

```diff
@@ -112,3 +112,35 @@ variable "variables_set" {
   default     = []
   sensitive   = true
 }
+
+variable "self_hosted_linux_agent_enable" {
+  type        = bool
+  description = "Self hosted linux agent enable"
+  default     = false
+}
+
+variable "pool_configuration" {
+  type = object({
+    agent_name     = string
+    auto_provision = optional(bool)
+    auto_update    = optional(bool)
+  })
+  description = "Objects to configure pool"
```

Comment:
Update description to - 'Object with parameters to configure self-hosted agent pool'